timelock-agent/src/main/java/com/palantir/timelock/paxos/TimeLockAgent.java

```diff
@@ -65,24 +68,41 @@ public TimeLockAgent(TimeLockInstallConfiguration install,
         this.paxosResource = PaxosResource.create(install.paxos().dataDirectory().toString());
         this.leadershipCreator = new PaxosLeadershipCreator(install, runtime, registrar);
         this.lockCreator = new LockCreator(runtime, deprecated);
-        this.timestampCreator = new PaxosTimestampCreator(paxosResource,
-                PaxosRemotingUtils.getRemoteServerPaths(install),
-                PaxosRemotingUtils.getSslConfigurationOptional(install).map(SslSocketFactories::createSslSocketFactory),
-                JavaSuppliers.compose(TimeLockRuntimeConfiguration::paxos, runtime));
+        this.timestampCreator = getTimestampCreator();
         this.timelockCreator = install.asyncLock().useAsyncLockService()
                 ? new AsyncTimeLockServicesCreator(leadershipCreator, install.asyncLock())
                 : new LegacyTimeLockServicesCreator(leadershipCreator);
     }
 
+    private TimestampCreator getTimestampCreator() {
+        TsBoundPersisterConfiguration timestampBoundPersistence = install.timestampBoundPersistence();
+        if (PaxosTsBoundPersisterConfiguration.class.isInstance(timestampBoundPersistence)) {
+            return getPaxosTimestampCreator();
+        } else if (DatabaseTsBoundPersisterConfiguration.class.isInstance(timestampBoundPersistence)) {
+            return new DbBoundTimestampCreator(
+                    ((DatabaseTsBoundPersisterConfiguration) timestampBoundPersistence)
+                            .keyValueServiceConfig());
+        } else {
+            throw new RuntimeException(String.format("Unknown TsBoundPersisterConfiguration found %s",
+                    timestampBoundPersistence.getClass()));
+        }
+    }
+
+    private PaxosTimestampCreator getPaxosTimestampCreator() {
```

Comment:
This seems a bit funky, in that it looks like we ignore the Paxos configuration provided for timestamp bound storage. So if I'm node A, and I configure the root level Paxos cluster to [A, B, C] _and_ the paxos ts bound persister to have cluster [A, D, E], I will still end up using [A, B, C] for both purposes. See comment on the Paxos persister configuration.

Reply:
(note: We don't have to worry about this problem any more, now that we've removed the configuration of the Paxos cluster in a Paxos persister configuration.)